packages/js/src/dashboard/components/google-site-kit-connection-widget.js

```diff
@@ -0,0 +1,158 @@
+ * @param {string} installUrl The URL to install Site Kit.
+ * @param {string} activateUrl The URL to activate Site Kit.
+ * @param {string} setupUrl The URL to setup Site Kit.
+ *
+ * @returns {Object} The button and stepper props.
+ */
+const getButtonAndStepperProps = ( active, installed, setup, connected, installUrl, activateUrl, setupUrl ) => {
+	let buttonProps;
+	let currentStep;
+	let isComplete;
+
+	switch ( true ) {
+		case ( ! installed ):
+			currentStep = 1;
+			isComplete = false;
+			buttonProps = {
+				children: __( "Install Site Kit by Google", "wordpress-seo" ),
+				as: "a",
+				href: installUrl,
+			};
+			break;
+		case ( ! active ):
+			currentStep = 2;
+			buttonProps = {
+				children: __( "Activate Site Kit by Google", "wordpress-seo" ),
+				as: "a",
+				href: activateUrl,
+			};
+			break;
+		case ( ! setup ):
+			currentStep = 3;
+			buttonProps = {
+				children: __( "Set up Site Kit by Google", "wordpress-seo" ),
+				as: "a",
+				href: setupUrl,
+			};
+			break;
+		case ( ! connected ):
+			currentStep = 4;
+			buttonProps = { children: __( "Connect Site Kit by Google", "wordpress-seo" ) };
+			break;
+		case connected:
+			isComplete = true;
+			currentStep = 4;
+			buttonProps = { children: "Take a quick tour" };
+			break;
+	}
+	return { buttonProps, currentStep, isComplete };
+};
+
+/**
+ * The google site kit connection guide widget.
+ *
+ * @param {boolean} installed Whether the plugin is installed.
+ * @param {boolean} setup Whether the setup is complete.
+ * @param {boolean} active Whether the feature is active.
+ * @param {boolean} connected Whether the connection is active.
+ * @param {boolean} installed Whether the plugin is installed.
+ * @param {string} activateUrl The URL to activate Site Kit.
+ * @param {string} installUrl The URL to install Site Kit.
+ * @param {function} onRemove The function to call when the widget is removed.
+ * @param {function} onRemovePermanently The function to call when the widget is removed permanently.
+ *
+ * @returns {JSX.Element} The widget.
+ */
+export const GoogleSiteKitConnectionWidget = ( {
+	installUrl,
+	activateUrl,
+	setupUrl,
+	connected,
+	active,
+	setup,
+	installed,
+	onRemove,
+	onRemovePermanently,
+} ) => {
+	const steps = [
+		{ label: __( "INSTALL", "wordpress-seo" ) },
+		{ label: __( "ACTIVATE", "wordpress-seo" ) },
+		{ label: __( "SET UP", "wordpress-seo" ) },
+		{ label: __( "CONNECT", "wordpress-seo" ) },
+	];
+	const [ isMenuOpen, toggleMenuOpen ] = useToggleState( false );
+	const linkParams = useSelect( select => select( "@yoast/general" ).selectLinkParams(), [] );
+	const learnMorelink = addQueryArgs( "https://yoa.st/google-site-kit-learn-more", linkParams );
+
+	const { buttonProps, currentStep, isComplete } = getButtonAndStepperProps(
+		active, installed, setup, connected, installUrl, activateUrl, setupUrl );
+	return <Paper className="yst-@container yst-grow yst-max-w-screen-sm yst-p-8 yst-shadow-md yst-relative">
+		<button
```

Comment:
This could be the `tertiary` variant (but then it would be color primary), or maybe it is time to add another variant to the UI library button?
I think this is not good for our focus styles, spacing etc:
![image](https://github.com/user-attachments/assets/86bc0382-c3c3-49c1-a11e-2d42c38345de)

In fact, the whole popup/menu thing is something the other widgets will also need.
We could chose to extract something at that point, but might be good to think about it a bit at least?

Reply:
Created the DropdownMenu